modules/core/exported/channel.go

```diff
@@ -34,6 +34,14 @@ type PacketI interface {
 // Acknowledgement defines the interface used to return
 // acknowledgements in the OnRecvPacket callback.
 type Acknowledgement interface {
+	// Success determines if the IBC application state should be persisted when handling `RecvPacket`.
+	// During `OnRecvPacket` IBC application callback execution, all state changes are held in a cache store and committed later.
+	//
+	// Note 1: IBC application callback events are always persisted so long as `RecvPacket` succeeds without error.
+	//
+	// Note 2: This method is independent of application level success/error which is encoded in the acknowledgement bytes in a protocol specific way.
+	//
+	// See https://github.com/cosmos/ibc-go/blob/main/docs/ibc/apps.md for further explanations.
```

Comment:
side note, we've been trying to avoid links to `main` in code comments but we are in the process of migrating documentation to docusaurus https://github.com/cosmos/ibc-go/pull/3511 so I think we can handle updating this out of band to a versioned doc link :)

Reply:
Makes sense. But what exactly do you prefer to do here and now? Link to a tag or commit? Leave it as is?

Reply:
Happy to leave as is, or you can drop a tag in if you want. Maybe https://github.com/cosmos/ibc-go/blob/v7.0.0/docs/ibc/apps.md

Reply:
Done. Version pinning is always a good idea.